### PR TITLE
feat: Add SemaphoreCI Connector

### PR DESCRIPTION
**Pull request by saviogl**
## Description

Took a stab at adding a SemaphoreCI source and faros converter

- [Semaphore's API](https://docs.semaphoreci.com/reference/api-v1alpha/)

Tested against the latest Faros CE

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#590 ](Add SemaphoreCI Source Connector) 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    organization: string;
+    repository: string;
+  } {
+    const urls = {
+      organization: '',
+      repository: '',
+    };
+
+    switch (source) {
+      case RepoSource.GITHUB:
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts:46]: Do we need to handle  `RepoSource.BITBUCKET`?

File: destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      case PipelineResult.Failed:
+        return {category: BuildStateCategory.Failed, detail};
+      case PipelineResult.Passed:
+        return {category: BuildStateCategory.Success, detail};
+      default:
+        return {category: BuildStateCategory.Custom, detail};
+    }
+  }
+
+  static nullifyDate(isoDate: string): string | undefined {
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts:110]: Why is this needed? Maybe add some short docs

File: sources/semaphoreci-source/resources/spec.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        "default": []
+      },
+      "branches": {
+        "order": 4,
+        "type": "array",
+        "items": {
+          "type": "string"
+        },
+        "title": "VCS Branch Name (Optional)",
+        "description": "Comma separated list of branch names to filter the dataset. If not set, all data will be pulled from all branches.",
```
Comment (ypc-faros) [inline sources/semaphoreci-source/resources/spec.json:49]: Remove "Comma separated"

**Review by ypc-faros — COMMENTED** [review]
Awesome 👏 👏 👏  ! Just a couple of comments

File: destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts (hunk 1 above)
Comment (saviogl) [reply destinations/airbyte-faros-destination/src/converters/semaphoreci/common.ts:46]: @ypc-faros Semaphore just added bitbucket support and I'm not fully aware how this is exposed in the api. I've set the stage but I currently lack additional context to fully add the support at the moment

**Comment by ypc-faros** [comment]
Thank you Sávio!

There's a test failure (`destinations/airbyte-faros-destination/test/converters/semaphoreci.test.ts`) due to a missing test resource file (`semaphoreci/all-streams.log`)

**Review by ypc-faros — APPROVED** [review]
Thank you @saviogl 🥇 !!!
